test/renate/memory/test_buffer.py

```diff
@@ -12,19 +12,74 @@
     ReservoirBuffer,
     SlidingWindowBuffer,
 )
+from renate.memory.buffer import _make_storage, _insert_data_point, _get_data_point
+
+
+class NestedTensorDataset(torch.utils.data.Dataset):
+    def __init__(self, nested_tensors):
+        self._nested_tensors = nested_tensors
+        self._length = None
+        self._get_len(nested_tensors)
```

Comment:
I would assign this directly to `self._length` instead of setting it in `_get_len()`

Reply:
`_get_len()` is recursively checking all tensors for consistent lengths. I think this is the most concise way to do this.

Reply:
it's as concise but you get rid of the side effect and the getter actually returns a value:

```
    def _get_len(self, nested_tensors, expected_length=None) -> int:
        if isinstance(nested_tensors, torch.Tensor):
            length = nested_tensors.size(0)
            assert length == expected_length or expected_length is None
            return length
        elif isinstance(nested_tensors, tuple):
            for t in nested_tensors:
                expected_length = self._get_len(t, expected_length)
            return expected_length
        elif isinstance(nested_tensors, dict):
            for t in nested_tensors.values():
                expected_length = self._get_len(t, expected_length)
            return expected_length
```

Reply:
Yep, fine with me. I changed it.